package/src/simulator_sw_0701/simu.py

```diff
@@ -255,41 +252,42 @@ def preparation(self, simulation_time, output_folder, sequences_file, Metadata_f
     def each_repeat(self, Metadata_file, simulation_time, output_folder,
                     sequences_file, p, r, c, MB_DRM, redo_count, switch):
 
-        # generate random seed
+        # generate a random seed
         switch = False
-        rng = stochastic_function(self.child_states \
-                                  [self.simulation_time_list.index(simulation_time) * \
-                                   self.redo_number + redo_count])
+        rng = stochastic_function(self.child_states 
+                                  [self.simulation_time_list.index(simulation_time) * 
+                                  self.redo_number + redo_count])
         
         progeny_pool_size_list = [] 
 
         # simulate in g generations until a viral rebound or suppression is achieved
         for generation in range(1, self.g + 1):
-            progeny_list, progeny_pool_size, recombined_mutated_sequences_file = \
-            each_generation(sequences_file, self.snv, self.rec, self.score_info, \
+            progeny_list, progeny_pool_size, recombined_mutated_sequences_file = 
+            each_generation(sequences_file, self.snv, self.rec, self.score_info, 
                             self.ref, self.treatment, p, r, c, MB_DRM, self.R, rng)
             progeny_pool_size_list.append(progeny_pool_size)
-            note = str(generation) + ':' + str(progeny_pool_size)
+            note = f"{generation}: {progeny_pool_size}"
             metadata(Metadata_file, note)
 
             # check
-            if len(progeny_pool_size_list) >= 3 and progeny_pool_size < 100 and \
+            if len(progeny_pool_size_list) >= 3 and progeny_pool_size < 100 and 
             progeny_pool_size_list[-2] < 100 and progeny_pool_size_list[-3] < 100:
```

Comment:
I think multi-line conditional statements like this should be within parenthesis.  Moreover, since your statement is already multi-line, I prefer to increase readability by making each comparison its own line as follows:
```suggestion
            if (len(progeny_pool_size_list) >= 3
            and progeny_pool_size < 100
            and progeny_pool_size_list[-2] < 100
            and progeny_pool_size_list[-3] < 100):
```